tests/main.spec.js

```diff
@@ -397,5 +398,40 @@ describe('StepZilla', () => {
         expect(enzymeWrapper.find('.progtrckr').childAt(2).hasClass('progtrckr-doing')).to.be.true;
       });
     });
+    
+    describe('onStepChange: not null use case', () => {
+        let onStepChange;
+        let enzymeWrapper;
+        
+        beforeEach(() => {
+            onStepChange = sinon.spy();
+            enzymeWrapper = setup(3, {
+                startAtStep: 1,
+                onStepChange
+            }).enzymeWrapper;
+        })
+    
+        it('should call onStepChange when clicked to next step', (done) => {
+            
+            enzymeWrapper.find('.footer-buttons #next-button').simulate('click');
+    
+            // click above is promise driven so it's async, setTimeout is probably not the best way to do this but it will do for now
+            setTimeout(() => {
```

Comment:
can't we use without setTimeout?

Reply:
as a user of this component I agree with @Aapzu and while I respect unit tests and also not a huge fan of setTimeout I dont think this is reason to block such an important feature, maybe the most important part of this component.

Reply:
Well, I think it'd be quite hypocritical not to accept this because of my usage of a method also used earlier in the code.

Reply:
@Aapzu check the conversation, I am forking my own version for my ideal api :)